File: 01. Leetcode/faangdsa/sachin/arrays/06/ValidAnagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()){
            return false;
        }

        Map<Character, Integer> counter = new HashMap<>();
        for (char i: s.toCharArray()){
            counter.put(i, counter.getOrDefault(i, 0)+1);
        }
        for (char i: t.toCharArray()){
            counter.put(i, counter.getOrDefault(i, 0)-1);
        }
        for (int val: counter.values()){
            if (val != 0){
                return false;
            }
        }

        return true;
    }
}

File: 01. Leetcode/faangdsa/sachin/arrays/06/validAnagram.ts
function isAnagram(s: string, t: string): boolean {
    if (s.length != t.length){
        return false;
    }

    const counter = new Map<string, number>();
    for (let i = 0; i<s.length; i++){
        counter.set(s[i], (counter.get(s[i]) || 0) +1)
    }
    for (let i = 0; i<s.length; i++){
        counter.set(t[i], (counter.get(t[i]) || 0) - 1)
    }   
    for (const val of counter.values()) {
        if (val !== 0) {
            return false;
        }
    }

    return true;
};